### PR TITLE
Adjust representation of arrays and maps in expressions

### PR DESCRIPTION
**Pull request by m-kuhn**
Just pushing this out here as a proposal, help texts and tests will need to be adjusted

array representation:

```
[ 1, 2, 3 ]
```

before:

```
array: 1, 2, 3
```

map representation:

```
{ 'one': 1, 'two': 2, 3: 'three' }
```

before

```
map: one: 1, two: 2, 3: three
```

### CODE REVIEW

**Review by Gustry — APPROVED** [review]
Yes, good move I think ;-)

**Comment by nirvn** [comment]
Nice. If you ask me, you could even get rid of the "array" and "map" prefix.

**Comment by elpaso** [comment]
maybe a dumb observation, but why don't we try to stick to a well known representation like JSON or Python (which is almost the same btw), it would make it easier to copy/paste from/to code when needed.

**Comment by m-kuhn** [comment]
+1

  * to remove prefixes
  * to go with JSON, what should be changed?

**Comment by elpaso** [comment]
JSON map would be `{ "one": 1, "two": 2, "three": 3, "four": "4 as a string" }`  (basically you need to quote the keys, and the values if they are not numeric or other JSON objects)

Lists/arrays are good: `[1, 2, 3, "a string"]`

**Comment by elpaso** [comment]
not sure if that can be useful for en/de-coding, but I think we are already including QJsonWrapper for oauth (now in core):  https://github.com/qgis/QGIS/tree/master/src/auth/oauth2/qjsonwrapper

File: src/gui/qgsexpressionbuilderwidget.cpp [hunk 1]
```diff
@@ -628,9 +628,9 @@ void QgsExpressionBuilderWidget::txtExpressionString_textChanged()
   if ( text.isEmpty() )
   {
     lblPreview->clear();
-    lblPreview->setStyleSheet( QLatin1String( "" ) );
```
Comment (nyalldawson) [inline src/gui/qgsexpressionbuilderwidget.cpp]: We really should add `QLatin1String( "" )` and `QStringLiteral( "" )` to the banned keywords test

**Pull request merged**